lib/serviceBuilder.js

```diff
@@ -48,16 +48,18 @@ function statusCodeError(statusCode, allowedStatusCodes, message) {
 function setBodyRelatedHeaders(body, options) {
   const isJSON = !Buffer.isBuffer(body) && typeof body !== 'string' && !(body && typeof body.pipe === 'function')
   if (isJSON) {
-    options.headers = options.headers || {}
-    options.headers['content-type'] = 'application/json;charset=utf-8'
+    options.headers = { 'content-type': 'application/json;charset=utf-8', ...options.headers }
   }
 
   if (!body) {
-    options.headers = options.headers || {}
-    options.headers['content-length'] = '0'
+    options.headers = { 'content-length': '0', ...options.headers }
```

Comment:
You changed the behavior of the function on this line.
Before your change, the `content-length` header was always set to `0`;
now the `content-length` header is set to `0` if the `options.headers` object does not contains it.

Is this change wanted? Or we can risk some regression.
You can adjust it by changing this line into:
```javascript
options.headers = { ...options.headers, `content-length': '0' }
```

Reply:
You're completely right, this was not wanted. I forgot to reverse the order